<html>
<head><meta charset="utf-8"><title>launching jupyter servers on login nodes via jupyterhub · jupyterlab-hub · Zulip Chat Archive</title></head>
<h2>Stream: <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/index.html">jupyterlab-hub</a></h2>
<h3>Topic: <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html">launching jupyter servers on login nodes via jupyterhub</a></h3>

<hr>


<head><link href="http://127.0.0.1:4000/style.css" rel="stylesheet"></head>

<a name="29466"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/29466" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Anderson Banihirwe <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#29466">(Apr 13 2021 at 22:02)</a>:</h4>
<p>Just wondering if anyone has had trouble with launching Jupyter server on <strong>Login nodes</strong> via the new JupyterHub (<a href="https://jupyterhub.hpc.ucar.edu/" target="_blank" title="https://jupyterhub.hpc.ucar.edu/">https://jupyterhub.hpc.ucar.edu/</a>). ?</p>
<p>All my attempts result in this error </p>
<p><a href="../../../user_uploads/2/e4/GT1o5XZdKTtDYWZmAYaT5wuQ/Screen-Shot-2021-04-13-at-2.47.59-PM.png" target="_blank" title="Screen-Shot-2021-04-13-at-2.47.59-PM.png">Screen-Shot-2021-04-13-at-2.47.59-PM.png</a></p>
<div class="message_inline_image"><a href="../../../user_uploads/2/e4/GT1o5XZdKTtDYWZmAYaT5wuQ/Screen-Shot-2021-04-13-at-2.47.59-PM.png" target="_blank" title="Screen-Shot-2021-04-13-at-2.47.59-PM.png"><img src="../../../user_uploads/2/e4/GT1o5XZdKTtDYWZmAYaT5wuQ/Screen-Shot-2021-04-13-at-2.47.59-PM.png"></a></div>



<a name="29467"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/29467" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Anderson Banihirwe <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#29467">(Apr 13 2021 at 22:03)</a>:</h4>
<p><strong>Note</strong>: Everything works fine when I used the Batch Job options</p>



<a name="29468"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/29468" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Jared Baker <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#29468">(Apr 13 2021 at 22:04)</a>:</h4>
<p>haven't seen that one before; Both casper and cheyenne?</p>



<a name="29470"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/29470" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Anderson Banihirwe <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#29470">(Apr 13 2021 at 22:05)</a>:</h4>
<blockquote>
<p>haven't seen that one before; Both casper and cheyenne?</p>
</blockquote>
<p>Both</p>



<a name="29471"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/29471" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Cecile Hannay <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#29471">(Apr 13 2021 at 22:05)</a>:</h4>
<p>I was also having an issue with:  <a href="https://jupyterhub.hpc.ucar.edu/" target="_blank" title="https://jupyterhub.hpc.ucar.edu/">https://jupyterhub.hpc.ucar.edu/</a><br>
<a href="../../../user_uploads/2/96/PBJmMI__b7XV9tiIvrxkO9xg/Screen-Shot-2021-04-13-at-2.01.02-PM.png" target="_blank" title="Screen-Shot-2021-04-13-at-2.01.02-PM.png">Screen-Shot-2021-04-13-at-2.01.02-PM.png</a></p>
<div class="message_inline_image"><a href="../../../user_uploads/2/96/PBJmMI__b7XV9tiIvrxkO9xg/Screen-Shot-2021-04-13-at-2.01.02-PM.png" target="_blank" title="Screen-Shot-2021-04-13-at-2.01.02-PM.png"><img src="../../../user_uploads/2/96/PBJmMI__b7XV9tiIvrxkO9xg/Screen-Shot-2021-04-13-at-2.01.02-PM.png"></a></div><p>BUt these might be unrelated.</p>



<a name="29472"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/29472" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Anderson Banihirwe <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#29472">(Apr 13 2021 at 22:12)</a>:</h4>
<blockquote>
<p>BUt these might be unrelated. </p>
</blockquote>
<p><span class="user-mention" data-user-id="60">@Cecile Hannay</span>, you are right. Your issue may have to do with corrupted browser cache &amp; cookies. Try accessing that URL from a <strong>private window</strong> or a different browser to confirm that things work when you use a fresh session.</p>



<a name="29473"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/29473" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Anderson Banihirwe <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#29473">(Apr 13 2021 at 22:14)</a>:</h4>
<p>If everything works in a clean session, I'd then try flushing locally cached browser data (clear your browser cookies and cache) .</p>



<a name="29474"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/29474" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Haiying Xu <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#29474">(Apr 13 2021 at 22:16)</a>:</h4>
<p><span class="user-mention" data-user-id="13">@Anderson Banihirwe</span>   Maybe restart your laptop to see if the problem will be solved.</p>



<a name="29475"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/29475" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Jared Baker <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#29475">(Apr 13 2021 at 22:18)</a>:</h4>
<p><span class="user-mention" data-user-id="13">@Anderson Banihirwe</span>  Can you deactivate your profile startup scripts temporarily?</p>



<a name="29476"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/29476" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Anderson Banihirwe <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#29476">(Apr 13 2021 at 22:26)</a>:</h4>
<blockquote>
<p><span class="user-mention silent" data-user-id="13">Anderson Banihirwe</span>  Can you deactivate your profile startup scripts temporarily?</p>
</blockquote>
<p>That works..... What's the culprit in my profile startup scripts?</p>



<a name="29477"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/29477" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Jared Baker <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#29477">(Apr 13 2021 at 22:32)</a>:</h4>
<p>usually something that prints to stdout. I thought I caught all of them, but apparently not. Mind if I look through them?</p>



<a name="29478"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/29478" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Anderson Banihirwe <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#29478">(Apr 13 2021 at 22:36)</a>:</h4>
<blockquote>
<p>Mind if I look through them? </p>
</blockquote>
<p>Not at all...</p>



<a name="29496"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/29496" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Jared Baker <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#29496">(Apr 13 2021 at 23:02)</a>:</h4>
<p>can you put it back for a quick moment?</p>



<a name="29500"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/29500" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Anderson Banihirwe <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#29500">(Apr 14 2021 at 00:05)</a>:</h4>
<p>Put it back breaks it again <span aria-label="frown" class="emoji emoji-1f641" role="img" title="frown">:frown:</span></p>



<a name="30335"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/30335" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Nan Rosenbloom <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#30335">(Apr 27 2021 at 03:59)</a>:</h4>
<p><span class="user-mention" data-user-id="13">@Anderson Banihirwe</span>  Hi Anderson - I'm having trouble getting onto jupyterhub from any of the options (casper/cheyenne, login/batch).  Any ideas what to try?</p>



<a name="30336"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/30336" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Rosie Fisher <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#30336">(Apr 27 2021 at 07:53)</a>:</h4>
<p>Me too <span class="user-mention" data-user-id="96">@Nan Rosenbloom</span></p>



<a name="30352"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/30352" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Jared Baker <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#30352">(Apr 27 2021 at 13:25)</a>:</h4>
<p><span class="user-mention" data-user-id="96">@Nan Rosenbloom</span> can you try again at your convenience?</p>



<a name="30357"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/30357" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Jared Baker <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#30357">(Apr 27 2021 at 13:46)</a>:</h4>
<p><span class="user-mention" data-user-id="161">@Rosie Fisher</span> Would you also please try again at your convenience?</p>



<a name="30358"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/30358" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Rosie Fisher <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#30358">(Apr 27 2021 at 13:51)</a>:</h4>
<p>Just tried and it still failed.  Message is<br>
 "Spawn failed: The Jupyter batch job has disappeared while pending in the queue or died immediately after starting."</p>



<a name="30360"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/30360" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Nan Rosenbloom <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#30360">(Apr 27 2021 at 15:05)</a>:</h4>
<p><span class="user-mention" data-user-id="154">@Jared Baker</span> I got failures on my 2 casper attempts; success on the cheyenne node login.  Last night the cheyenne node would hang for 15-20 minutes before failing.  Progress!</p>



<a name="30361"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/30361" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Matt Long <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#30361">(Apr 27 2021 at 15:07)</a>:</h4>
<p>For what it's worth, I just spawned two servers one on Casper Login and one on Casper Batch.</p>



<a name="30363"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/30363" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Jared Baker <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#30363">(Apr 27 2021 at 15:19)</a>:</h4>
<p>The Cheyenne login node was the one I felt was the lowest hanging fruit on that. I'll work on Casper ones next.</p>



<a name="30386"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/30386" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Jared Baker <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#30386">(Apr 27 2021 at 16:10)</a>:</h4>
<p><span class="user-mention" data-user-id="96">@Nan Rosenbloom</span> Can you try stopping your server and log out and back in then try a Casper session again?</p>



<a name="30387"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/30387" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Rosie Fisher <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#30387">(Apr 27 2021 at 16:23)</a>:</h4>
<p>Mine is working now, on cheyenne at least.</p>



<a name="30389"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/30389" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Rosie Fisher <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#30389">(Apr 27 2021 at 16:41)</a>:</h4>
<p>Oh, I spoke too soon. It is hanging...</p>



<a name="30399"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/30399" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Jared Baker <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#30399">(Apr 27 2021 at 17:22)</a>:</h4>
<p><span class="user-mention" data-user-id="161">@Rosie Fisher</span> I sent you a PM. Can you respond there?</p>



<a name="31075"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/31075" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Will Wieder <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#31075">(May 10 2021 at 17:01)</a>:</h4>
<p>anyone else having issue logging onto casper using jupyter-forward?  My browser is prompting for a password/token.  I eventually landed on the right password, but curious why I got this prompt after weeks of not being asked?</p>



<a name="31092"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/31092" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Will Wieder <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#31092">(May 10 2021 at 17:44)</a>:</h4>
<p>Now I'm getting this error trying to execute jupyter-forward from the command line.<br>
<span aria-label="cross mark" class="emoji emoji-274c" role="img" title="cross mark">:cross_mark:</span> Couldn't find jupyter executable with: 'conda activate  core.py:97<br>
           lens-py &amp;&amp; sh -c "command -v jupyter"'</p>



<a name="31093"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/31093" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Anderson Banihirwe <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#31093">(May 10 2021 at 17:49)</a>:</h4>
<p><span class="user-mention" data-user-id="67">@Will Wieder</span>, </p>
<p>which version of <code>jupyter-forward</code> are you using? </p>
<div class="codehilite"><pre><span></span>jupyter-forward --version
</pre></div>



<a name="31097"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/31097" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Will Wieder <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#31097">(May 10 2021 at 17:56)</a>:</h4>
<p>Jupyter Forward CLI Version: 2021.3.3</p>



<a name="31098"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/31098" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Anderson Banihirwe <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#31098">(May 10 2021 at 18:00)</a>:</h4>
<p>Using the python/ipython interpreter, do the following  code snippets work for you?</p>
<div class="codehilite"><pre><span></span><span class="n">In</span> <span class="p">[</span><span class="mi">2</span><span class="p">]:</span> <span class="kn">from</span> <span class="nn">jupyter_forward</span> <span class="kn">import</span> <span class="n">RemoteRunner</span>

<span class="n">In</span> <span class="p">[</span><span class="mi">3</span><span class="p">]:</span> <span class="n">runner</span> <span class="o">=</span> <span class="n">RemoteRunner</span><span class="p">(</span><span class="s1">&#39;abanihi@casper.ucar.edu&#39;</span><span class="p">)</span>
<span class="o">************************************************************************************************</span> <span class="n">Authentication</span> <span class="o">*************************************************************************************************</span>
<span class="n">Authenticating</span> <span class="n">user</span> <span class="p">(</span><span class="n">abanihi</span><span class="p">)</span> <span class="kn">from</span> <span class="nn">client</span> <span class="p">(</span><span class="n">cisl</span><span class="o">-</span><span class="n">sublimity</span><span class="p">)</span> <span class="n">to</span> <span class="n">remote</span> <span class="n">host</span> <span class="p">(</span><span class="n">casper</span><span class="o">.</span><span class="n">ucar</span><span class="o">.</span><span class="n">edu</span><span class="p">)</span>
<span class="n">Token_Response</span><span class="p">:</span>
<span class="err">✅</span> <span class="n">The</span> <span class="n">client</span> <span class="ow">is</span> <span class="n">authenticated</span> <span class="n">successfully</span>

<span class="n">In</span> <span class="p">[</span><span class="mi">4</span><span class="p">]:</span> <span class="n">runner</span><span class="o">.</span><span class="n">run_command</span><span class="p">(</span><span class="s1">&#39;which python&#39;</span><span class="p">)</span>
<span class="n">which</span> <span class="n">python</span>
<span class="o">/</span><span class="n">glade</span><span class="o">/</span><span class="n">work</span><span class="o">/</span><span class="n">abanihi</span><span class="o">/</span><span class="n">opt</span><span class="o">/</span><span class="n">miniconda</span><span class="o">/</span><span class="nb">bin</span><span class="o">/</span><span class="n">python</span>
<span class="n">Out</span><span class="p">[</span><span class="mi">4</span><span class="p">]:</span> <span class="o">&lt;</span><span class="n">Result</span> <span class="n">cmd</span><span class="o">=</span><span class="s1">&#39;which python&#39;</span> <span class="n">exited</span><span class="o">=</span><span class="mi">0</span><span class="o">&gt;</span>

<span class="n">In</span> <span class="p">[</span><span class="mi">5</span><span class="p">]:</span> <span class="n">runner</span><span class="o">.</span><span class="n">run_command</span><span class="p">(</span><span class="s1">&#39;which conda&#39;</span><span class="p">)</span>
<span class="n">which</span> <span class="n">conda</span>
<span class="o">/</span><span class="n">glade</span><span class="o">/</span><span class="n">work</span><span class="o">/</span><span class="n">abanihi</span><span class="o">/</span><span class="n">opt</span><span class="o">/</span><span class="n">miniconda</span><span class="o">/</span><span class="nb">bin</span><span class="o">/</span><span class="n">conda</span>
<span class="n">Out</span><span class="p">[</span><span class="mi">5</span><span class="p">]:</span> <span class="o">&lt;</span><span class="n">Result</span> <span class="n">cmd</span><span class="o">=</span><span class="s1">&#39;which conda&#39;</span> <span class="n">exited</span><span class="o">=</span><span class="mi">0</span><span class="o">&gt;</span>

<span class="n">In</span> <span class="p">[</span><span class="mi">6</span><span class="p">]:</span> <span class="n">runner</span><span class="o">.</span><span class="n">run_command</span><span class="p">(</span><span class="s1">&#39;which jupyter&#39;</span><span class="p">)</span>
<span class="n">which</span> <span class="n">jupyter</span>
<span class="o">/</span><span class="n">glade</span><span class="o">/</span><span class="n">work</span><span class="o">/</span><span class="n">abanihi</span><span class="o">/</span><span class="n">opt</span><span class="o">/</span><span class="n">miniconda</span><span class="o">/</span><span class="nb">bin</span><span class="o">/</span><span class="n">jupyter</span>
<span class="n">Out</span><span class="p">[</span><span class="mi">6</span><span class="p">]:</span> <span class="o">&lt;</span><span class="n">Result</span> <span class="n">cmd</span><span class="o">=</span><span class="s1">&#39;which jupyter&#39;</span> <span class="n">exited</span><span class="o">=</span><span class="mi">0</span><span class="o">&gt;</span>
</pre></div>



<a name="31099"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/31099" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Will Wieder <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#31099">(May 10 2021 at 18:23)</a>:</h4>
<p>sorry, where do I run these code snippets?</p>



<a name="31100"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/31100" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Anderson Banihirwe <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#31100">(May 10 2021 at 18:24)</a>:</h4>
<p>From your local computer.... To launch the interpreter, run  <code>python</code></p>



<a name="31101"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/31101" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Will Wieder <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#31101">(May 10 2021 at 18:25)</a>:</h4>
<p>it seems to get stuck here </p>
<blockquote>
<blockquote>
<blockquote>
<p>from jupyter_forward import RemoteRunner<br>
runner = RemoteRunner('<a href="mailto:wwieder@ucar.edu" title="mailto:wwieder@ucar.edu">wwieder@ucar.edu</a>')<br>
<strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><strong> Authentication </strong><strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><br>
[12:24:10] Authenticating user (wwieder) from client (cgdm-begbie) to core.py:66<br>
           remote host (<a href="http://ucar.edu" target="_blank" title="http://ucar.edu">ucar.edu</a>)</p>
</blockquote>
</blockquote>
</blockquote>



<a name="31102"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/31102" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Will Wieder <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#31102">(May 10 2021 at 18:25)</a>:</h4>
<p>Traceback (most recent call last):<br>
  File "&lt;stdin&gt;", line 1, in &lt;module&gt;<br>
  File "&lt;string&gt;", line 11, in __init__<br>
  File "/Users/wwieder/miniconda3/lib/python3.8/site-packages/jupyter_forward/core.py", line 71, in __post_init__<br>
    self.session.open()<br>
  File "/Users/wwieder/miniconda3/lib/python3.8/site-packages/fabric/connection.py", line 636, in open<br>
    self.client.connect(**kwargs)<br>
  File "/Users/wwieder/miniconda3/lib/python3.8/site-packages/paramiko/client.py", line 349, in connect<br>
    retry_on_signal(lambda: sock.connect(addr))<br>
  File "/Users/wwieder/miniconda3/lib/python3.8/site-packages/paramiko/util.py", line 283, in retry_on_signal<br>
    return function()<br>
  File "/Users/wwieder/miniconda3/lib/python3.8/site-packages/paramiko/client.py", line 349, in &lt;lambda&gt;<br>
    retry_on_signal(lambda: sock.connect(addr))<br>
TimeoutError: [Errno 60] Operation timed out</p>



<a name="31103"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/31103" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Anderson Banihirwe <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#31103">(May 10 2021 at 18:26)</a>:</h4>
<p>Okay.... I don't know what's going on so far <span aria-label="grinning" class="emoji emoji-1f600" role="img" title="grinning">:grinning:</span> </p>
<blockquote>
<p>Jupyter Forward CLI Version: 2021.3.3</p>
</blockquote>
<p>Do you mind upgrading <code>jupyter-forward</code> to the latest version <code>v2021.4.13</code>, and tyring again ?</p>



<a name="31105"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/31105" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Will Wieder <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#31105">(May 10 2021 at 18:28)</a>:</h4>
<p>sorry, didn't include casper on my loging runner = RemoteRunner('<a href="mailto:wwieder@casper.ucar.edu" title="mailto:wwieder@casper.ucar.edu">wwieder@casper.ucar.edu</a>')<br>
<strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong> Authentication <strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><br>
[12:26:46] Authenticating user (wwieder) from client (cgdm-begbie) to remote host (<a href="http://casper.ucar.edu" target="_blank" title="http://casper.ucar.edu">casper.ucar.edu</a>)       core.py:66<br>
Token_Response: <br>
[12:26:53] <span aria-label="check" class="emoji emoji-2705" role="img" title="check">:check:</span> The client is authenticated successfully</p>



<a name="31106"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/31106" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Will Wieder <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#31106">(May 10 2021 at 18:29)</a>:</h4>
<p>but then I cannot run any of the <code>runner.run_command()</code> lines.</p>



<a name="31107"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/31107" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Will Wieder <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#31107">(May 10 2021 at 18:32)</a>:</h4>
<p>update line this?  <br>
conda update jupyter-forward</p>
<p>PackageNotInstalledError: Package is not installed in prefix.<br>
  prefix: /Users/wwieder/miniconda3<br>
  package name: jupyter-forward</p>



<a name="31108"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/31108" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Will Wieder <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#31108">(May 10 2021 at 18:34)</a>:</h4>
<p>ah, it's pip<br>
pip install jupyter-forward -U</p>



<a name="31109"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/31109" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Will Wieder <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#31109">(May 10 2021 at 18:36)</a>:</h4>
<p>update is done, but now this error</p>
<blockquote>
<blockquote>
<blockquote>
<p>from jupyter_forward import RemoteRunner<br>
runner = RemoteRunner('<a href="mailto:wwieder@casper.ucar.edu" title="mailto:wwieder@casper.ucar.edu">wwieder@casper.ucar.edu</a>')<br>
<strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong> Authentication <strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><strong>*</strong><br>
Authenticating user (wwieder) from client (cgdm-begbie) to remote host (<a href="http://casper.ucar.edu" target="_blank" title="http://casper.ucar.edu">casper.ucar.edu</a>)<br>
Exception: Error reading SSH protocol banner<br>
Traceback (most recent call last):<br>
  File "/Users/wwieder/miniconda3/lib/python3.8/site-packages/paramiko/transport.py", line 2211, in _check_banner<br>
    buf = self.packetizer.readline(timeout)<br>
  File "/Users/wwieder/miniconda3/lib/python3.8/site-packages/paramiko/packet.py", line 380, in readline<br>
    buf += self._read_timeout(timeout)<br>
  File "/Users/wwieder/miniconda3/lib/python3.8/site-packages/paramiko/packet.py", line 622, in _read_timeout<br>
    raise socket.timeout()<br>
socket.timeout</p>
</blockquote>
</blockquote>
</blockquote>
<p>During handling of the above exception, another exception occurred:</p>
<p>Traceback (most recent call last):<br>
  File "/Users/wwieder/miniconda3/lib/python3.8/site-packages/paramiko/transport.py", line 2039, in run<br>
    self._check_banner()<br>
  File "/Users/wwieder/miniconda3/lib/python3.8/site-packages/paramiko/transport.py", line 2215, in _check_banner<br>
    raise SSHException(<br>
paramiko.ssh_exception.SSHException: Error reading SSH protocol banner</p>
<p>Traceback (most recent call last):<br>
  File "/Users/wwieder/miniconda3/lib/python3.8/site-packages/paramiko/transport.py", line 2211, in _check_banner<br>
    buf = self.packetizer.readline(timeout)<br>
  File "/Users/wwieder/miniconda3/lib/python3.8/site-packages/paramiko/packet.py", line 380, in readline<br>
    buf += self._read_timeout(timeout)<br>
  File "/Users/wwieder/miniconda3/lib/python3.8/site-packages/paramiko/packet.py", line 622, in _read_timeout<br>
    raise socket.timeout()<br>
socket.timeout</p>
<p>During handling of the above exception, another exception occurred:</p>
<p>Traceback (most recent call last):<br>
  File "&lt;stdin&gt;", line 1, in &lt;module&gt;<br>
  File "&lt;string&gt;", line 11, in __init__<br>
  File "/Users/wwieder/miniconda3/lib/python3.8/site-packages/jupyter_forward/core.py", line 69, in __post_init__<br>
    self.session.open()<br>
  File "/Users/wwieder/miniconda3/lib/python3.8/site-packages/fabric/connection.py", line 636, in open<br>
    self.client.connect(**kwargs)<br>
  File "/Users/wwieder/miniconda3/lib/python3.8/site-packages/paramiko/client.py", line 406, in connect<br>
    t.start_client(timeout=timeout)<br>
  File "/Users/wwieder/miniconda3/lib/python3.8/site-packages/paramiko/transport.py", line 660, in start_client<br>
    raise e<br>
  File "/Users/wwieder/miniconda3/lib/python3.8/site-packages/paramiko/transport.py", line 2039, in run<br>
    self._check_banner()<br>
  File "/Users/wwieder/miniconda3/lib/python3.8/site-packages/paramiko/transport.py", line 2215, in _check_banner<br>
    raise SSHException(<br>
paramiko.ssh_exception.SSHException: Error reading SSH protocol banner</p>



<a name="31110"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/31110" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Will Wieder <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#31110">(May 10 2021 at 18:39)</a>:</h4>
<p>but I'm able to launch a notebook now?!  No idea why, but it seems to be working (for now).</p>



<a name="31113"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/31113" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Anderson Banihirwe <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#31113">(May 10 2021 at 19:11)</a>:</h4>
<blockquote>
<p>but then I cannot run any of the <code>runner.run_command()</code> lines.</p>
</blockquote>
<p>I think this is available in <code>v2021.4.13</code> or later</p>



<a name="31114"></a>
<h4><a href="https://zulip2.cloud.ucar.edu#narrow/stream/16-jupyterlab-hub/topic/launching%20jupyter%20servers%20on%20login%20nodes%20via%20jupyterhub/near/31114" class="zl"><img src="../../../assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Anderson Banihirwe <a href="http://127.0.0.1:4000/html/stream/16-jupyterlab-hub/topic/launching.20jupyter.20servers.20on.20login.20nodes.20via.20jupyterhub.html#31114">(May 10 2021 at 19:12)</a>:</h4>
<blockquote>
<p>but I'm able to launch a notebook now?!  No idea why, but it seems to be working (for now).</p>
</blockquote>
<p>Okay... Glad it's working even though we have no idea what went wrong before the sucessful attempt.</p>



<hr><p>Last updated: May 16 2025 at 17:14 UTC</p>
</html>